python/tchannel/handler.py

```diff
@@ -108,17 +104,17 @@ def handle_pre_call(self, message_id, message, connection):
             connection.send_error(
                 ErrorCode.bad_request,
                 e.message,
-                message_id,
+                message.id,
             )
-        except Exception:
+        except Exception as e:
             connection.send_error(
                 ErrorCode.unexpected,
-                "An unexpected error has occurred!",
-                message_id,
+                "An unexpected error has occurred!" + e.message,
```

Comment:
Probably not a good idea to include service-internal error messages in external responses. Skip doing this. Also, drop the `!`.

(If we were to include the message, concatenation like that doesn't make for a pretty error message.)